brainiak/reprsimil/brsa.py

```diff
@@ -464,51 +873,78 @@ def _prepare_data_XY(self, X, Y, D, F):
+        n_X0 = X0.shape[1]
+        X0TX0, X0TDX0, X0TFX0 = self._make_templates(D, F, X0, X0)
+        XTX0, XTDX0, XTFX0 = self._make_templates(D, F, X, X0)
+        X0TY, X0TDY, X0TFY = self._make_templates(D, F, X0, Y)
+
+        return X0TX0, X0TDX0, X0TFX0, XTX0, XTDX0, XTFX0, \
+            X0TY, X0TDY, X0TFY, X0, X_base, n_X0, idx_DC
+
+    def _merge_DC_to_base(self, X_DC, X_base, no_DC):
+        """ Merge DC components X_DC to the baseline time series
+            X_base (By baseline, this means any fixed nuisance
+            regressors not updated during fitting, including DC
+            components and any nuisance regressors provided by
+            the user.
+            X_DC is always in the first few columns of X_base.
+        """
+        if X_base is not None:
+            res0 = np.linalg.lstsq(X_DC, X_base)
             if not no_DC:
-                res0 = np.linalg.lstsq(X_base, X0)
                 if not np.any(np.isclose(res0[1], 0)):
-                    # No columns in X0 can be explained by the
+                    # No columns in X_base can be explained by the
                     # baseline regressors. So we insert them.
-                    X0 = np.concatenate((X_base, X0), axis=1)
+                    X_base = np.concatenate((X_DC, X_base), axis=1)
+                    idx_DC = np.arange(0, X_DC.shape[1])
                 else:
-                    logger.warning('Provided regressors for non-interesting '
+                    logger.warning('Provided regressors for uninteresting '
                                    'time series already include baseline. '
                                    'No additional baseline is inserted.')
+                    idx_DC = np.where(np.isclose(res0[1], 0))[0]
+            else:
+                idx_DC = np.where(np.isclose(res0[1], 0))[0]
         else:
             # If a set of regressors for non-interested signals is not
             # provided, then we simply include one baseline for each run.
-            X0 = X_base
+            X_base = X_DC
+            idx_DC = np.arange(0, X_base.shape[1])
             logger.info('You did not provide time series of no interest '
-                        'such as DC component. One trivial regressor of'
-                        ' DC component is included for further modeling.'
+                        'such as DC component. Trivial regressors of'
+                        ' DC component are included for further modeling.'
                         ' The final covariance matrix won''t '
-                        'reflet them.')
-        n_base = X0.shape[1]
-        X0TX0, X0TDX0, X0TFX0 = self._make_templates(D, F, X0, X0)
-        XTX0, XTDX0, XTFX0 = self._make_templates(D, F, X, X0)
-        X0TY, X0TDY, X0TFY = self._make_templates(D, F, X0, Y)
-
-        return X0TX0, X0TDX0, X0TFX0, XTX0, XTDX0, XTFX0, \
-            X0TY, X0TDY, X0TFY, X0, n_base
+                        'reflet these components.')
```

Comment:
Does "won't" work? It is within a string, so I thought won''t works as a escape for special character.